Generals/Code/GameEngine/Source/Common/RTS/TunnelTracker.cpp

```diff
@@ -332,12 +332,15 @@ void TunnelTracker::updateFullHealTime()
 			continue;
 
 		const ContainModuleInterface* contain = tunnelObj->getContain();
-		const TunnelContain* tunnelContain = dynamic_cast<const TunnelContain*>(contain); // Downcast is not ideal
+		if (!contain->isTunnelContain())
+			continue;
+
+		const TunnelContain* tunnelContain = static_cast<const TunnelContain*>(contain); // Downcast is not ideal
 		if (!tunnelContain)
```

Comment:
This null test is either too late or superfluous because `contain` was already dereferenced before this.

I think it is safe to earlier just do

`DEBUG_ASSERTCRASH(contain != NULL, ("contain module is NULL"));`